File: src/main/java/com/hypherionmc/sdlink/core/events/VerificationEvent.java
/*
 * This file is part of sdlink-core, licensed under the MIT License (MIT).
 * Copyright HypherionSA and Contributors
 */
package com.hypherionmc.sdlink.core.events;

import com.hypherionmc.craterlib.core.event.CraterEvent;
import com.hypherionmc.sdlink.core.accounts.MinecraftAccount;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * @author HypherionSA
 * Events that get triggered when the bot verification list changes
 */
public class VerificationEvent {

    @Getter
    @RequiredArgsConstructor
    public static class PlayerVerified extends CraterEvent {
        private final MinecraftAccount account;
    }

    @Getter
    @RequiredArgsConstructor
    public static class PlayerUnverified extends CraterEvent {
        private final MinecraftAccount account;
    }

}

File: Common/src/main/kotlin/com/hypherionmc/sdlinkrw/api/events/VerificationEvent.kt
package com.hypherionmc.sdlinkrw.api.events

import com.hypherionmc.craterlib.core.event.CraterEvent
import com.hypherionmc.sdlinkrw.api.accounts.MinecraftAccount

/**
 * @author HypherionSA
 *
 * Events that get fired when a player is verified or unverified.
 */
class VerificationEvent {

    /**
     * Fired when a player is verified.
     */
    class PlayerVerified(val account: MinecraftAccount): CraterEvent()

    /**
     * Fired when a player is unverified.
     */
    class PlayerUnverified(val account: MinecraftAccount): CraterEvent()
}